collections/_article/migrate-animation-from-godot-4-0-to-4-3.md

```diff
@@ -0,0 +1,317 @@
+
+This is an option mainly to control cases where the total blending weight is `0`.
+
+**For example:**
+
+When comparing two animations, there are cases where only one has a track and the other does not.
+
+![animations with unmatched track count 1](/storage/blog/migrate-animation-from-godot-4-0-to-4-3/anim1.webp)
+![animations with unmatched track count 2](/storage/blog/migrate-animation-from-godot-4-0-to-4-3/anim2.webp)
+
+When `animation1` and `animation2` are blended so that `animation2` is 100%, the scale track will have the total blending weight is `0` in the result.
+
+**We can expect two possible behaviors in here:**
+
+1. Do nothing
+2. Apply default value
+
+The behavior of 1 is that AnimationTree on 3.x, and AnimationPlayer was forced to perform between 3.x, 4.0 and 4.1.
+
+AnimationTree adopted the behavior of 2 as of 4.0 because it was necessary to process additive blending correctly.
+
+In additive blending, cases where the total weight of blending is `total_weight < 1` or `total_weight > 1` must be interpolated from the default value. This means that if the total blending weight is `0`, the animation will be set to the default value. This means that if one animation has no track in the blending, the default value will be applied, and the result will look like performing a reset.
+
+This reset behavior is also convenient with the retargeting introduced in 4.0.
+
+Some models may have extra bones between the retargeted bones. In this case, when blending occurs with an animation that does not have extra bones, the appearance will be inconsistent if the extra bones are not reset.
+
+**For example:**
+
+Model A has an animation "look up" with 3 bone tracks.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/retarget1.mp4?1" type="video/mp4">
+</video>
+
+Model B has an animation "bow" with 4 bone tracks.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/retarget2.mp4?1" type="video/mp4video/mp4">
+</video>
+
+Then, let's play these animations sequentially without resetting.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/migrate-animation-from-godot-4-0-to-4-3/retarget3.mp4?1" type="video/mp4">
+</video>
+
+It creates a condition where the appearance of the animations do not match between them. So, as you can see from the results, the 3D model pipeline required that resetting behavior.
+
+However, it was not only beneficial, but also caused issues in some cases. Especially frequently reported were cases using `AnimationNodeStateMachine` with 2D project.
+
+If there is no crossfade in the `AnimationNodeStateMachine` transition, it appears to be just an animation is switched, but internally it is blending, which causes the resetting. This made confusing because it was appeared to be doing the same thing as AnimationPlayer, but it was behaving differently.
```

Comment:
```suggestion
If there is no crossfade in the `AnimationNodeStateMachine` transition, the animation appears to suddenly switch without blending, but internally it is blending, which causes the resetting. This was confusing because it was appeared to be doing the same thing as AnimationPlayer, but it was behaving differently.
```